internal/worker/ondiskname/ondiskname.go

```diff
@@ -0,0 +1,53 @@
+package ondiskname
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+var ErrInvalidOnDiskName = errors.New("invalid on-disk VM name")
+
+const prefix = "orchard"
+
+type OnDiskName struct {
+	Name string
+	UID  string
+}
+
+func New(name string, uid string) OnDiskName {
+	return OnDiskName{
+		Name: name,
+		UID:  uid,
+	}
+}
+
+func Parse(s string) (OnDiskName, error) {
+	splits := strings.Split(s, "-")
```

Comment:
I think this logic fails for VMs that have `-` in their name. I think logic can be simplified in terms that we know a VM starts with `orchard-` and end with 36 suffix `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`.

Reply:
See https://github.com/cirruslabs/orchard/pull/70/commits/ea9fb45d5cb7b5158bd9f00d60efc08c3bf1a95f.

Reply:
Logic will still fail on name with two hypens like `test-vm--1`. There is also no validation that the UUID is the right format. I think removing prefix and getting the last 36 characters for UID will be error prone and simpler to read.

Reply:
I don't think the logic will fail per se, it will just prevent empty parts, which probably makes no sense to prevent anyway.

I've removed this check in 96019fa and updated the test.

Reply:
Do we need to validate the UUID though?

Reply:
> Do we need to validate the UUID though?

I don't think we need to, the only purpose of `OnDiskName` is to detect Orchard-managed VMs and split their VM's names and UIDs properly.

Reply:
ok